### DISCUSSION TITLE
Some tuning /adds

### DISCUSSION THREAD

**Discussion by arkahys** [stefanhendriks/Dune-II---The-Maker#358] [Ideas]
I'm french so, I'll express me my best.
Thanks to this work on the first RTS I played. Cult game !! So many thanks !
 I read your todo list. maybe some idea can avoid changes or adjust meanwhile adding more stuff.
Would talk about what make game unplaisant to play for me , as i'm a fan of RTS , and the game is , not very attractive in gameplay:

- When placing structure if a unit is selected, it will move (as cursor is move) on placement. 

- If cursos is out of battlefieldscreen : "screen move" and "cancel unit selected" doesn't work (then line before occurs).
Right click resolve many other selection/cancel/action problem.

- Move screen can't be done by touch border screen  (who is appreciable, when your cursor is on the edge, you can't move screen by mouse, you have to comeback cursor to use right click), LEFT+RIGHT click can be used too. I find the acceleration move mode not reactive to play quickly, a drag screen ( or drag layer like smartphone to become a sheep =)   ) is usefull.

- Can't click and move (unit) on minimap.
- Can't place quickly structures : 
"auto select freshly new construction(on cursor) when no selection and building menu"
then when placing small slab, you can enqueue them, place... waiting, place, waiting, place.. etc... without travel all screen with mouse.
You can make a "strating build" by enqueue building and waiting each finish.
In this interresting gameplay feature, adding "greyed locked structure based on enqueued building" allowed to be add in queue (and launch only if building is available to build at the building moment)

- there is no graduations on power (and silos) bar, then it is hard to know if following building will outage your power.

Units who move at their wish is really, really, (I think it is the worth things make me not replay a skirmish) annoying (units and harvester)
Add a unit/structure menu can allow features, waiting better implement (as original game) :
move mode : waiting or Auto  Then by default unit don't move at all. then I can say to harvester to not to go to raffinery when full to lost spicy because silos full and units don't cross the map while I'm occuppied to buildings.
Retreat : go to first friend building (or rafinery for harvester).
Call carryall  : Go to the actually moving point (move,repear, raffinery or retreat) with carryall, can add many tactics. As in the original game, put a repair facility without ground acces make carryall take unit, repear AND carry back the unit at the carry point). The original game have so many usefull fonction !
Upgrade : for turret.
Rally point (for product factory) : Newly unit built will go to the rally point.
Desactive(power) : a little shortage of power, if you make bad plans on the last building to build a fanpower.
option for ornithopter  and carryall on Hight-tech factory : comeback or Fly (use it as hangar and repair facility, if building constructed)
  
Units move jerkly. may units calculate the after go point before stop on a square, IF true THEN don't slow speed.
 Balance moves by : 
 big nerf of acceleration of heavy units .
 slowdown light units max speed.
 increase rotate of all, a little bit on heavy (no on siege), and much on light,
 All units should fire more quickly (or anticipate a straight line fire). Because nerf of heavy with their low rotating speed (turret and body), they don't fire on a light who cross in straight line.
 This will make a good gameplay, more stone sheet chisel, lights units keep interrests even in late game.
 
Tuning :
Make all factory product each enqueued units (actually product quickly. With desactive option, you can don't use a specific factory).
Sonic tank seems to attack only the selected units. But in original game attack all the straight line friend as ennemieSS (Was cool.. and Uncool.. but the deal).
Buildings seems to be constructed very quiclky compared to units. You can have last game units in somes minutes...

Hoping that give you ideas !

**Comment by arkahys** [comment]
Although annoying... I replay again and again, haaa, childhood !!!  ;-).
Sorry, I answered too quickly, I didn't take the time to know what was really interesting to say and miss issues already posted.
 I don't see the scroll by edge window because really too slow.
Can you increase default and add a parameter in d2tm.cfg ?

I will make a new post next days to propose an usefull improvement of dune 2 to be enjoy (my idea and what is yours), because many indy game I follow, and contribute, spend time in "little improvement" to rewrite (and time) few time later to counterbalance a new trial/need.

**Comment by stefanhendriks** [comment]
About `Units move jerkly. may units calculate the after go point before stop on a square, IF true THEN don't slow speed.` -> this is how Dune 2 units move. Square by square (no fluent movement) so this is intentional.

**Reply by arkahys** [reply]
Yes, I realize, that they don't accelerate or slow. I don't know why I Thought their speed change. I make some test , seems to be because units stop on each cell before go to  next. I Presume by a "wait for verify condition of free next cell". I haven't watched code yet. I'm not  ease in C.
Do you have corrected the original source code, or mostly write it ?

**Reply by stefanhendriks** [reply]
D2TM is its own engine. So all self made and not based on the original source code
